packages/react-components/src/shared/src/checkable.ts

```diff
@@ -1,19 +1,24 @@
 import { createContext, useContext } from "react";
 import { isNil } from "lodash";
 
-export const CheckableContext = createContext();
+interface CheckableContextProps {
+    checkedValue?: boolean;
+    [x: string]: any;
+}
+
+export const CheckableContext = createContext<CheckableContextProps>(null);
 
 export function useCheckableContext() {
     const context = useContext(CheckableContext);
 
     if (!isNil(context)) {
-        return [context, true];
+        return [context, true] as const;
```

Comment:
I would naively think you don't to do `as const` and simply set the return type of the function as an array

Reply:
This is used for most of the custom hooks you use. 

const [state, setState] = useState(boolean)


without the as const, useState would return an array of boolean and functions

(boolean | (value: boolean) => void)[]
so you wouldn't be able to use setState(false) since setState could be a boolean according to typescript

but in fact, we know its an array of 2 elements, the first one is a bool, and the second one is a function. 

this blog shows the 2 way of typing this. i prefer the 2nd option

https://fettblog.eu/typescript-react-typeing-custom-hooks/#option-2%3A-as-const